examples/externalContactsDemo/script.js

```diff
@@ -0,0 +1,269 @@
+            if (contactId) {
+                clientApp.externalContacts.showExternalContactProfile(contactId);
+            }
+        });
+    document
+        .querySelector('.manual-example #orgButton')
+        .addEventListener('click', function () {
+            let externalOrganizationId = document.querySelector('#org-id').value.trim();
+            if (externalOrganizationId) {
+                clientApp.externalContacts.showExternalOrganizationProfile(externalOrganizationId);
+            }
+        });
+
+    let redirectUrl = window.location.origin;
+    if (!redirectUrl) {
+        redirectUrl = window.location.protocol + '//' + window.location.host;
+    }
+    redirectUrl += window.location.pathname;
+
+    // Authenticate with PureCloud
+    let authenticated = false;
+    let userDataAcquired = false;
+
+    client
+        .loginImplicitGrant(clientId, redirectUrl, {
+            state: 'pcEnvironment=' + pcEnvironment
+        })
+        .then(function () {
+            updateProgressBar(100);
+            authenticated = true;
+            return new platformClient.UsersApi().getUsersMe({
+                expand: ['authorization']
+            });
+        })
+        .then(function (profileData) {
+            userDataAcquired = true;
+            // Check if the current user will be able to view external contacts
+            let permissions = profileData.authorization.permissions;
+            if (checkPermission(permissions, 'externalContacts:contact:view')) {
+                requestExternalContacts().then(data => {
+                    let entities = data.entities;
+                    // Build table with data from api call
+                    for (let i = 0; i < entities.length; i++) {
+                        let hasOrg = !!entities[i].externalOrganization;
+                        document.getElementById('tableBody').insertAdjacentHTML('beforeend',
+                            `<tr>
+                                <td>
+                                    <a id="entity-${i}">${entities[i].lastName}, ${entities[i].firstName}</a>
+                                </td>
+                                <td>
+                                    <a id="org-${i}">${hasOrg ? entities[i].externalOrganization.id : ''}</a>
+                                </td>
+                            </tr>`
+                        );
+                        // Attach listeners to entities to call sdk methods
+                        document.getElementById(`entity-${i}`).addEventListener('click', function () {
+                            clientApp.externalContacts.showExternalContactProfile(entities[i].id);
+                        });
+                        if (hasOrg) {
+                            document.getElementById(`org-${i}`).addEventListener('click', function () {
+                                clientApp.externalContacts.showExternalOrganizationProfile(entities[i].externalOrganization.id);
```

Comment:
`clientApp.externalContacts` is undefined here as the module seems to no longer be in the clientApp closure since I separated this example out into its own directory. Not sure why that would be the case. It worked before and the test for Modules passes (`'should have instances of all the modules'`) in `ClientAppSpec.js` 🤷